### PR TITLE
Activity stream

### PR DESCRIPTION
**Pull request by andrewconner**
Real time activity stream using Play's `Iteratees` and `Enumerator` with WebSockets.

### CODE REVIEW

**Comment by eishay** [comment]
We'll do a tech/talk meeting tomorrow where Andrew will educate us on how this works

File: shoebox/app/com/keepit/common/analytics/ActivityStream.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import org.joda.time.DateTime
+import com.keepit.inject._
+
+@Singleton
+class ActivityStream {
+  implicit val timeout = Timeout(1 second)
+
+  lazy val default = Akka.system.actorOf(Props[ActivityStreamActor])
+
+  def newStream(): Future[(Iteratee[JsValue,_],Enumerator[JsValue])] = {
```
Comment (eishay) [inline shoebox/app/com/keepit/common/analytics/ActivityStream.scala:31]: space after comma - its free
my old eyes are not parsing that well

File: shoebox/app/com/keepit/controllers/BookmarksController.scala [hunk 2]
```diff
@@ -297,4 +302,25 @@ object BookmarksController extends FortyTwoController {
...
+
+  private def addToActivityStream(user: User, bookmark: Bookmark) = {
+    val social = inject[DBConnection].readOnly { implicit session =>
+      inject[SocialUserInfoRepo].getByUser(user.id.get).headOption.map(_.socialId.id).getOrElse("")
+    }
+
+    val json = Json.obj(
+      "user" -> Json.obj(
+        "id" -> user.id.get.id,
+        "name" -> s"${user.firstName} ${user.lastName}",
```
Comment (eishay) [inline shoebox/app/com/keepit/controllers/BookmarksController.scala:314]: wowow string formatting made easy

**Comment by eishay** [comment]
nice!

File: shoebox/app/views/adminRTEventViewer.scala.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      var seconds = date.getSeconds()
+
+      if (hours   < 10) {hours   = "0"+hours;}
+      if (minutes < 10) {minutes = "0"+minutes;}
+      if (seconds < 10) {seconds = "0"+seconds;}
+      var time    = hours+':'+minutes+':'+seconds;
+      return time;
+    }
+      
+    var WS = window['MozWebSocket'] ? MozWebSocket : WebSocket
```
Comment (2is10) [inline shoebox/app/views/adminRTEventViewer.scala.html:33]: oooooooooooooooo

**Pull request merged**